deps/wazuh_testing/wazuh_testing/process_resource_monitoring/src/process_resource_monitoring/monitor.py

```diff
@@ -0,0 +1,289 @@
+            - PID: pid of the process.
+            - CPU(%): cpu percent of the process. It maybe greater than 100% if the process uses multiple threads.
+            - VMS: Virtual Memory Size.
+            - RSS: Resident Set Size.
+            - USS: Unique Set Size.
+            - PSS: Proportional Set Size.
+            - SWAP: memory of the process in the swap space.
+            - FD: file descriptors opened by the process.
+            - Read_Ops: read operations.
+            - Write_Ops: write operations.
+            - Disk_Read: Bytes read by the process.
+            - Disk_Written: Bytes written by the process.
+
+        Args:
+            proc (psutil.Process): psutil object with the data of the process.
+
+        Returns:
+            (Dict[str, Any]): Dictionary containing the data of the process.
+        """
+
+        def unit_conversion(number: int) -> float:
+            """Converts bytes to the specified unit in the constructor.
+
+            Args:
+                number (int): number in bytes to be converted.
+
+            Returns:
+                (float): number of the unit chosen in the constructor.
+            """
+            return number / (1024 ** self._DATA_UNITS[self._value_unit])
+
+        # Pre-initialize the info dictionary. If there's a problem while taking metrics of the binary (i.e. it crashed)
+        # the CSV will set all its values to 0 to easily identify if there was a problem or not
+        info = {
+            'Daemon': self._process_name,
+            'Version': self._version,
+            'Timestamp': datetime.now().strftime('%Y/%m/%d %H:%M:%S'),
+            'PID': self._pid,
+            'CPU(%)': 0.0,
+            f'VMS({self._value_unit})': 0.0,
+            f'RSS({self._value_unit})': 0.0,
+            f'USS({self._value_unit})': 0.0,
+            f'PSS({self._value_unit})': 0.0,
+            f'SWAP({self._value_unit})': 0.0,
+            'FD': 0.0,
+            'Read_Ops': 0.0,
+            'Write_Ops': 0.0,
+            f'Disk_Read({self._value_unit})': 0.0,
+            f'Disk_Written({self._value_unit})': 0.0,
+            f'Disk_Read_Speed({self._value_unit}/s)': 0.0,
+            f'Disk_Write_Speed({self._value_unit}/s)': 0.0,
+        }
+
+        try:
+            with proc.oneshot():
+                info['CPU(%)'] = proc.cpu_percent(interval=None)
+                memory_data = proc.memory_full_info()
+                info[f'VMS({self._value_unit})'] = unit_conversion(memory_data.vms)
+                info[f'RSS({self._value_unit})'] = unit_conversion(memory_data.rss)
+                info[f'USS({self._value_unit})'] = unit_conversion(memory_data.uss)
+                info[f'PSS({self._value_unit})'] = unit_conversion(memory_data.pss)
+                info[f'SWAP({self._value_unit})'] = unit_conversion(memory_data.swap)
+                info['FD'] = proc.num_fds()
+
+                if self._platform == 'linux' or platform == 'win32':
```

Comment:
Psutil doesn't seem to support macOS that much. In the docs you can see how the availability for the method that is used in the script's conditional, Process.io_counters() is only available for Linux, BSD, Windows and AIX. Should we go through the trouble to make it work? It would probably take some time, but taking into account that the central components can only be installed on Linux then it seems reasonable to leave out those 4 out of 11 metrics (7 relevant metrics are allegedly still collected).

Reply:
True https://psutil.readthedocs.io/en/latest/#psutil.Process.io_counters, macOS is not supported